department-of-veterans-affairs/vets-api
# EDM-229/remove operating hours from sob

## Summary

- *This work is behind a feature toggle (flipper): YES/NO* No, removing code that was previously behind feature toggle :sob_updated_design
- *(Summarize the changes that have been made to the platform)*: Post 911 Statement of Benefits (SOB) successfully updated to 24/7 access [here](https://github.com/department-of-veterans-affairs/vets-api/commit/f9b931187f2525bf3c5187b37731286e9d1b6d16#diff-1c61846271b485d47d2bc559b628f97ac2cc1f78adb51f629412182c55b9de5eR31). Files and logic that referenced downtime were put behind feature toggle. Now that app has been running successfully 24/7, we're removing the code that's no longer relevant.
- *(If bug, how to reproduce)*
- *(What is the solution, why is this the solution?)*
  - Remove hardcoded operating hours and associated methods from Post911 controller and Lighthouse Benefits Education service
  - Remove errors and files associated with "outside working hours"
  - Update suite of spec tests that test for whether service is up or down
  - **Update backend statuses route, controller, and associated files**: The backend_statuses#show route appears to have been specifically created for our app. The #index action interfaces with pager duty, but as you can see in this [PR ](https://github.com/department-of-veterans-affairs/vets-api/commit/f9b931187f2525bf3c5187b37731286e9d1b6d16#diff-1c61846271b485d47d2bc559b628f97ac2cc1f78adb51f629412182c55b9de5eR31), the show action was specific to Post 911 GI Bill Status, and the `available?` method only checked hard-coded operating hours. It didn't actually reach out to upstream service. @stevenjcumming created the backend_status model during a refactor in that PR, but it still was tied specifically to those hardcoded operating hours.
- *(Which team do you work for, does your team own the maintenance of this component?)* Education Data Migration (EDM). yes to post 911 Statement of Benefits. No to the backend_statuses route and controller
- *(If introducing a flipper, what is the success criteria being targeted?)*

## Related issue(s)

- https://github.com/department-of-veterans-affairs/vets-api/pull/18657
- *Link to epic if not included in ticket*](https://jira.devops.va.gov/browse/EDM-229)

## Testing done

- *Describe what the old behavior was prior to the change*: Spec tests tested whether app was up or down according to hard coded operating hours
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*: Updated tests in previous PR after 24/7 access was enabled but used feature toggle in specs. Removing feature toggle to ensure behavior in test now matches prod
- *If this work is behind a flipper:* No flipper removed
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?* Feature has been tested before and after old operating hours. Removing flipper and will repeat testing steps of accessing app before and after old operating hours

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*: Post 911 Statement of Benefits

## Acceptance criteria

- [x]  Documentation has been updated (link to documentation): swagger for removed route
- [ x  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected

## Requested Feedback

As I mentioned above, @stevenjcumming updated the backend statuses controller and created the backend status model which I'm removing during a refactor. If the platform would like to maintain the backend_statuses#show route for a more universal purpose, we can leave code in. But it's all currently linked to whether GI Bill is up or down and so might need to be re-imagined
